### DISCUSSION TITLE
Should SumatraPDF be more lightweight?

### DISCUSSION THREAD

**Discussion by dungsaga** [sumatrapdfreader/sumatrapdf#3362] [General]
I read the blog post [Lessons learned from 15 years of SumatraPDF, an open source Windows app](https://blog.kowalczyk.info/article/2f72237a4230410a888acbfce3dc0864/lessons-learned-from-15-years-of-sumatrapdf-an-open-source-windows-app.html) and appreciate the effort to make it the most lightweight PDF reader today.

But then I recall that it was much more smaller when it was announced. I found some old 32-bit versions in FileHorse:
![image](https://user-images.githubusercontent.com/5884951/226223380-caf0199c-a286-4c1d-8bd8-2991111e463c.png)
The size of the installer increased from 1MB to 4MB to 6MB. The installed size would be around 2.2 or 2.5 times bigger.

I understand that SumatraPDF depends from libmupdf (which is 11MB in my 64-bit version) and libmupdf support many more file formats than Poppler. On top of that, SumatraPDF added support for many compressed archive formats and modern image formats.

I wonder if we can make it more lightweight like its previous versions by splitting some supports into DLL files that would be loaded dynamically. Would it be smaller if we use Poppler instead?

It's interesting to read the [SumatraPDF version history](https://www.sumatrapdfreader.org/docs/Version-history), in version 0.9 : "removed poppler rendering engine resulting in smaller program and updated to latest mupdf sources". At that time, the installer was only 1 MB.
The growth of mupdf contributes a lot to the size of SumatraPDF. Mupdf even added support for MOBI recently.

This is not a complain about SumatraPDF. I still love it and understand that today it comes with lots of functions and it's still smaller than other PDF readers (The little Foxit Reader is already bloated over 150 MB).

This is just a what-if kind of thought experiment I have when I look at the smallest PDF freemium editor : PicoPDF
![image](https://user-images.githubusercontent.com/5884951/226249243-760fe936-0d8a-4910-b803-85f4c85fb9d4.png)

**Comment by kjk** [comment]
Majority of the size are fonts. Without fonts, many documents won't render correctly.

The size is what it is. There is no easy fix and splitting into dlls is just fake accounting.

**Comment by dungsaga** [comment]
The size of the portable release is 33% smaller. It looks like all the libraries are static linked and packed (with mpress or something).
It's nice to have a leaner release for portable use.
![image](https://user-images.githubusercontent.com/5884951/227392519-86f68976-eb2f-4ee7-9838-2f5fd65a1bdf.png)

**Reply by GitHubRulesOK** [reply]
Considering MuPDF portable has less formats but is itself 41,326,080 bytes its amazing SumatraPDF is 1/3 rd as big
I grant you that pico is much smaller and was surprised until I remembered they download many support temporary functions all over the place. Several locations in users temp and in programdata too. still less diskspace but split files and usually empty folders even if report 0 bytes all use 4KB disk space sectors per filename even if  a short link to home.

**Reply by dungsaga** [reply]
Yes, they put some files in %APPDATA%/Roaming but the additional size is less than 1MB.
And the folder `C:\ProgramData\NCH Software\PicoPDF` is empty.
![image](https://user-images.githubusercontent.com/5884951/227428159-d0225090-6976-4948-810b-755f5727e469.png)
The features of PicoPDF is quite basic, so I think that they can keep the total size to be ~3.6MB
When SumatraPDF 0.1 was release, it was only a bit more than 1MB.

UPDATED:
when using OCR feature for the first time, it downloads TesseractOCR to %APPDATA%/Roaming (almost 30MB).
So the total size of additional components is 30.5MB
![image](https://user-images.githubusercontent.com/5884951/227434171-a873e3a4-a365-483c-aba2-d7de5a5eb555.png)

![image](https://user-images.githubusercontent.com/5884951/227430594-818349c3-97d5-4cd3-b515-a80eccac2793.png)
But for normal editing, PicoPDF can even work without network connection.

UPDATED AGAIN:
I was wrong. PicoPDF downloads some files when I open a PDF file.
http://www.nch.com.au/components/zlib1v3.exe
http://www.nch.com.au/components/libjpeg.exe
http://www.nch.com.au/components/littlecms.exe
And when I use OCR feature, it downloads http://www.nch.com.au/components/tesseractocr3.exe

So, you still need internet connection when you open a PDF for the first time.
And it's weird that these components are not listed in http://www.nch.com.au/components/
